Userland/Libraries/LibGfx/Font/VectorFont.h

```diff
@@ -52,6 +54,14 @@ class VectorFont : public RefCounted<VectorFont> {
     virtual u8 slope() const = 0;
     virtual bool is_fixed_width() const = 0;
     virtual bool has_color_bitmaps() const = 0;
+
+    [[nodiscard]] NonnullRefPtr<ScaledFont> scaled_font(float point_size) const;
+
+protected:
+    VectorFont();
+
+private:
+    mutable HashMap<float, NonnullRefPtr<ScaledFont>> m_scaled_fonts;
```

Comment:
It's weird layering-wise that VectorFont knows about ScaledFont, no? VectorFont is the base class of the classes that load vector fonts (ttf woff etc), while ScaledFont is the thing that abstracts over either bitmap fonts or vector fonts.

(I care a bit because I have a local branch that moves LibPDF to use VectorFont instead of Font/ScaledFont for $reasons. I don't know if I'll push that branch yet, but this seems weird layering-wise independent of that.)

Reply:
ScaledFont only ever abstracts over VectorFont. It's basically a reference to a VectorFont with specific metrics and a rasterized glyph cache.

Or perhaps I'm misunderstanding what you mean?

Reply:
We have Gfx::Font with subclasses BitmapFont and ScaledFont. ScaledFont implements Gfx::Font using VectorFont + subclasses.

So Gfx::Font are higher level and use VectorFont & co as implementation detail. So it's a bit weird to me if the lower level part knows about the higher level part. (It's all in one lib so it's technically fine, but IMHO still a bit weird.)

(ps: my phone keeps correcting VectorFont to VectorDon't which is funny to me.)

Reply:
Hmm, it's a bit more ambiguous IMO, `Gfx::VectorFont` is higher level in some contexts, since that's what `Gfx::Typeface` deals with. It's also what we construct from downloaded CSS fonts. In those cases, `Gfx::Font` is something we get from the `VectorFont`.

I'm sure this whole architecture could be improved, but for now this caching gives us a huge speedup by avoiding style recalculation on every style change. :)